src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSslVersion.cs

```diff
@@ -11,22 +11,17 @@ internal static partial class OpenSsl
     {
         private static Version s_opensslVersion;
 
-        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_SSLEayVersion")]
-        private static extern string OpenSslVersionDescription();
+        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_OpenSslVersionNumber")]
+        private static extern uint OpenSslVersionNumber();
 
         internal static Version OpenSslVersion
         {
             get
             {
                 if (s_opensslVersion == null)
                 {
-                    const string OpenSSL = "OpenSSL ";
-
-                    // Skip OpenSSL part, and get the version string of format x.y.z
-                    if (!Version.TryParse(OpenSslVersionDescription().AsSpan(OpenSSL.Length, 5).ToString(), out s_opensslVersion))
-                    {
-                        s_opensslVersion = new Version(0, 0, 0);
-                    }
+                    uint versionNumber = OpenSslVersionNumber();
+                    s_opensslVersion = new Version((int)(versionNumber >> 28), (int)((versionNumber >> 20) & 0xff), (int)((versionNumber >> 12) & 0xff));
```

Comment:
It made way more sense to parse the version number from the integer than from a string. The string parsing worked for OpenSSL and LibreSSL, but failed for BoringSSL. The native SSLeay function is present in all OpenSSL versions < 1.1, where it was replaced by OpenSSL_version_num returning the value in the same format.